sc-tools/sc-server/src/main.cpp

```diff
@@ -0,0 +1,104 @@
+/*
+ * This source file is part of an OSTIS project. For the latest info, see http://ostis.net
+ * Distributed under the MIT License
+ * (See accompanying file COPYING.MIT or copy at http://opensource.org/licenses/MIT)
+ */
+
+#include <boost/program_options.hpp>
+
+#include <iostream>
+#include <atomic>
+#include <thread>
+
+#include <sc-memory/sc_debug.hpp>
+#include <sc-memory/sc_memory.hpp>
+#include <sc-memory/utils/sc_signal_handler.hpp>
+
+#include "sc-config/sc_config.hpp"
+#include "sc_memory_config.hpp"
+
+#include "sc_server_factory.hpp"
+
+sc_int main(sc_int argc, sc_char * argv[])
+try
+{
+  boost::program_options::options_description options("Sc-server usage");
+  options.add_options()("help", "Display this message")(
+      "config,c", boost::program_options::value<std::string>(), "Path to configuration file")(
+      "host,h", boost::program_options::value<std::string>(), "Sc-server host name, ip-address")(
+      "port,p", boost::program_options::value<size_t>(), "Sc-server port")(
+      "extensions_path,e", boost::program_options::value<std::string>(), "Path to directory with sc-memory extensions")(
+      "repo_path,r", boost::program_options::value<std::string>(), "Path to kb.bin folder")(
+      "verbose,v", "Flag to don't save sc-memory state on exit")("clear", "Flag to clear sc-memory on start");
+
+  boost::program_options::variables_map vm;
+  boost::program_options::store(boost::program_options::command_line_parser(argc, argv).options(options).run(), vm);
+  boost::program_options::notify(vm);
+
+  if (vm.count("help"))
+  {
+    std::cout << options;
+    return EXIT_SUCCESS;
+  }
+
+  std::string configFile;
+  if (vm.count("config"))
+    configFile = vm["config"].as<std::string>();
+
+  ScParams serverParams;
+  std::vector<std::string> keys = {"host", "port"};
+  for (auto const & key : keys)
+  {
+    if (vm.count(key))
+      serverParams.insert({key, vm[key].as<std::string>()});
+  }
+
+  ScConfig config{configFile};
+  auto serverConfig = config["sc-server"];
+  for (auto const & key : *serverConfig)
+    serverParams.insert({key, serverConfig[key]});
+
+  ScParams memoryParams;
+  keys = {"extensions_path", "repo_path", "verbose", "clear"};
+  for (auto const & key : keys)
+  {
+    if (vm.count(key))
+      memoryParams.insert({key, vm[key].as<std::string>()});
+  }
```

Comment:
I am not sure if we'll need to do this often. If there's some repetition (or a chance that we could do it in agents) we can just add a function for this rather than creating a whole class

Reply:
Fixed